src/Main/User.java

```diff
@@ -0,0 +1,152 @@
+package Main;
+
+import java.util.Scanner;
+
+public class User {
+
+    private int turn = 1;
+    private String userAnswer;
+    private int userPredictionNumber;
+
+    public void promptUserForInput() {
+        Scanner input = new Scanner(System.in);
+
+        if (checkIfUserIsPredictor(turn) == true) {
+            System.out.println("You are the predictor.");
+        }
+
+        else {
+            System.out.println("You are not the predictor");
+        }
+
+        System.out.println("What is your input? ");
+        userAnswer = input.nextLine();
+
+        checkAllInput(getUserAnswer());
+    }
+
+    private void checkAllInput(String userAnswer) {
+        if (checkIfUserIsPredictor(turn) == true) {
+            checkInputLengthUserIsPredictor(getUserAnswer());
+            calculateAndCheckUserPredictionNumber(getUserAnswer());
+        }
+
+        else {
+            checkInputLengthUserNotPredictor(getUserAnswer());
+        }
+        checkIfOpenClosedExist(getUserAnswer());
+    }
+
+    private void checkInputLengthUserIsPredictor(String userAnswer) {
+        if (userAnswer.length() != 3) {
+            System.out.println("Bad input: correct input should be of the form CC3, " +
+                    "where the first two letters indicate [O]pen or [C]losed state for each hand, " +
+                    "followed by the prediction (0-4).");
+
+            promptUserForInput();  // start over
+        }
+    }
+
+    private void checkInputLengthUserNotPredictor(String userAnswer) {
+        if (userAnswer.length() < 2 || userAnswer.length() > 3) {
+            System.out.println("Bad input: correct input should be of the form OC, " +
+                    "where the letters indicate [O]pen or [C]losed state for each hand.");
+
+            promptUserForInput();
+        }
+
+        else if (userAnswer.length() == 3) {
+            System.out.println("Bad input: no prediction expected, you are not the predictor.");
+
+            promptUserForInput();
+        }
+    }
+
+    public boolean checkIfOpenClosedExist(String userAnswer) {  // made public instead of private for testing
+        boolean opedClosedExists = false;
+
+        if ((userAnswer.charAt(0) == 'O'
```

Comment:
Her er en anden måde at skrive det på der måske er lidt mere elegant 

```suggestion
       if (!"cCoO".indexOf( userAnswer.charAt(0) ) >= 0 || !"cCoO".indexOf( userAnswer.charAt(1) ) >= 0 || ) {
```

Reply:
Den her kan jeg sgu ikke lige få til at virke.. når jeg copy paster, beder den mig flytte "!" ud af parentesen (og slette det sidste ||), men når jeg gør det og kører programmet, så får jeg error message hver gang